src/groupchats.c

```diff
@@ -2214,6 +2206,8 @@ static void groupchat_onDraw(ToxWindow *self, Toxic *toxic)
         }
     }
 
+    pthread_mutex_unlock(&Winthread.lock);
+
```

Comment:
The locking is done inside the if-statement, so shouldn't that unlock call be inside the if-statement block too, right after the for-loop block ends?
```suggestion
        }

        pthread_mutex_unlock(&Winthread.lock);
    }

```